h2o-logger/src/test/java/water/logging/LoggerTest.java

```diff
@@ -0,0 +1,45 @@
+package water.logging;
+import org.junit.Test;
+
+import java.lang.reflect.Field;
+
+
+import static org.junit.Assert.*;
+
+public class LoggerTest {
+    @Test
+    public void FactoryTest() {
+        if (!isSlf4JAvailable()){
+            ILogger logger = LoggerFactory.getLogger(LoggerTest.class);
+            assertTrue(logger instanceof ConsoleLogger);
+        } else {
+            ILogger logger = LoggerFactory.getLogger(LoggerTest.class);
+            assertTrue(logger instanceof Slf4JLogger);
+        }
+    }
+
+    @Test (expected = ClassCastException.class)
+    public void ConstructorTest() throws IllegalAccessException, NoSuchFieldException {
+        
+        if (!isSlf4JAvailable()) {
+            Logger logger = new Logger(ConsoleLogger.class, LoggerTest.class);
+            Field f = logger.getClass().getDeclaredField("logger");
+            ConsoleLogger consoleLogger = (ConsoleLogger) f.get(logger);
```

Comment:
its not clear which one of these is supposed to throw the CCEx, this thest IMO will always pass